store/sqlstore/channel_store.go

```diff
@@ -474,42 +472,68 @@ func (s SqlChannelStore) saveChannelT(transaction *gorp.Transaction, channel *mo
 	return result
 }
 
+// Update writes the updated channel to the database.
+//
+// @see ChannelStoreExperimental for how this update propagates to the PublicChannels table.
 func (s SqlChannelStore) Update(channel *model.Channel) store.StoreChannel {
 	return store.Do(func(result *store.StoreResult) {
-		channel.PreUpdate()
-
-		if channel.DeleteAt != 0 {
-			result.Err = model.NewAppError("SqlChannelStore.Update", "store.sql_channel.update.archived_channel.app_error", nil, "", http.StatusBadRequest)
+		transaction, err := s.GetMaster().Begin()
+		if err != nil {
+			result.Err = model.NewAppError("SqlChannelStore.Update", "store.sql_channel.update.open_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
 			return
 		}
 
-		if result.Err = channel.IsValid(); result.Err != nil {
+		*result = s.updateChannelT(transaction, channel)
```

Comment:
I've factored out `*T` helper methods to isolate the actual SQL queries in play here. Strictly speaking, this isn't necessary, but it allows me to duplicate just the boilerplate begin/update/commit in the experimental store, reducing the chance of deviating if these methods are updated before we merge the two stores.

Where I've done this, there should be no other changes: unfortunately, the diff here isn't ideal.